virtcontainers/qemu.go

```diff
@@ -39,12 +41,15 @@ type CPUDevice struct {
 type QemuState struct {
 	Bridges []Bridge
 	// HotpluggedCPUs is the list of CPUs that were hot-added
-	HotpluggedVCPUs []CPUDevice
-	UUID            string
+	HotpluggedVCPUs  []CPUDevice
+	HotpluggedMemory int
```

Comment:
Why aren't you using a `memoryDevice` type instead here? Also, this would force this structure to be public.

Reply:
I did not change this because `memoryDevice.slot` is internal knowledge. Even if we have `sandbox.HotplugMemory`, we should not expose it. @sboeuf let me know if you think otherwise.

Reply:
Sorry, what I meant is that we could have something like `HotpluggedMemory []MemoryDevice` the same way we have `HotpluggedVCPUs []CPUDevice`.

Reply:
No, I don't think we need that. We only need to track how much memory has been hotplugged. Unlike CPU devices, we do not support memory hot unplug, nor need to unplug memory the same chunk as they were plugged in even if we support memory hot unplug in future.

Reply:
Fair enough !

Reply:
@bergwolf     Sorry I don't quite understand why you put "slot" field in struct MemoryDevice rather than struct QemuState , as an input from user will not involve "slot" field. When we hotplug memory device, we use a qmp command like "object_add ...  mem1 ...", in which "1" is memory slot number. We should store "slot" in QemuState which is hypervisor.json so that we could get memory slot number when we need to hotplug memory.